src/tests/HDFFile.cxx

```diff
@@ -118,155 +141,323 @@ class T_CommandHandler : public testing::Test {
     return true;
   }
 
+  /// Can supply pre-generated test data for a source on a topic to profile the
+  /// writing.
+  class SourceDataGen {
+  public:
+    string topic;
+    string source;
+    int seed = 0;
+    std::mt19937 rnd;
+    vector<FlatBufs::ev42::fb> fbs;
+    vector<FileWriter::Msg> msgs;
+    // Number of messages already fed into file writer during testing
+    size_t n_fed = 0;
+    /// Generates n test messages which we can later feed from memory into the
+    /// file writer.
+    void pregenerate(int n, int n_events_per_message) {
+      // LOG(4, "generating {} {}...", topic, source);
+      FlatBufs::ev42::synth synth(source, seed);
+      rnd.seed(seed);
+      for (int i1 = 0; i1 < n; ++i1) {
+        // Number of events per message:
+        // size_t n_ele = rnd() >> 24;
+        // Currently fixed, have to adapt verification code first.
+        size_t n_ele = n_events_per_message;
+        fbs.push_back(synth.next(n_ele));
+        auto &fb = fbs.back();
+        msgs.push_back(FileWriter::Msg{(char *)fb.builder->GetBufferPointer(),
+                                       (int32_t)fb.builder->GetSize()});
+        // LOG(4, "GetSize: {}", fb.builder->GetSize());
+      }
+    }
+  };
+
   static void data_ev42() {
```

Comment:
True.  This part has already changed quite a bit though in another upcoming feature branch, so I'd like to skip to break it up in this PR to avoid the rebase.